rosidl_parser/test/test_parser.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+
+import pytest
+
+from rosidl_parser.definition import Array
+from rosidl_parser.definition import BasicType
+from rosidl_parser.definition import BoundedSequence
+from rosidl_parser.definition import Constant
+from rosidl_parser.definition import IdlLocator
+from rosidl_parser.definition import Include
+from rosidl_parser.definition import Message
+from rosidl_parser.definition import Service
+from rosidl_parser.definition import String
+from rosidl_parser.definition import UnboundedSequence
+from rosidl_parser.definition import WString
+from rosidl_parser.parser import parse_idl_file
+
+MESSAGE_IDL_LOCATOR = IdlLocator(
+    pathlib.Path(__file__).parent, pathlib.Path('msg') / 'MyMessage.idl')
+SERVICE_IDL_LOCATOR = IdlLocator(
+    pathlib.Path(__file__).parent, pathlib.Path('srv') / 'MyService.idl')
+
+
+@pytest.fixture(scope='module')
+def message_idl_file():
+    return parse_idl_file(MESSAGE_IDL_LOCATOR)
+
+
+def test_message_parser(message_idl_file):
+    messages = message_idl_file.content.get_elements_of_type(Message)
+    assert len(messages) == 1
+
+
+def test_message_parser_includes(message_idl_file):
+    includes = message_idl_file.content.get_elements_of_type(Include)
+    assert len(includes) == 2
+    assert includes[0].locator == 'OtherMessage.idl'
+    assert includes[1].locator == 'pkgname/msg/OtherMessage.idl'
+
+
+def test_message_parser_constants(message_idl_file):
+    constants = message_idl_file.content.get_elements_of_type(Constant)
+    assert len(constants) == 5
+
+    constant = [c for c in constants if c.name == 'SHORT_CONSTANT']
+    assert len(constant) == 1
+    constant = constant[0]
+    assert isinstance(constant.type, BasicType)
+    assert constant.type.type == 'int16'
+    assert constant.value == -23
+
+    constant = [c for c in constants if c.name == 'UNSIGNED_LONG_CONSTANT']
+    assert len(constant) == 1
+    constant = constant[0]
+    assert isinstance(constant.type, BasicType)
+    assert constant.type.type == 'uint32'
+    assert constant.value == 42
+
+    constant = [c for c in constants if c.name == 'FLOAT_CONSTANT']
+    assert len(constant) == 1
+    constant = constant[0]
+    assert isinstance(constant.type, BasicType)
+    assert constant.type.type == 'float'
+    assert constant.value == 1.25
```

Comment:
It's probably safe to assume assigning the value `1.25` is the same code path as [`float('1.25')`](https://github.com/ros2/rosidl/blob/7647c6b042bd4c0d4135baf513c34948ec82e729/rosidl_parser/rosidl_parser/parser.py#L478), but [`pytest.approx()`](https://docs.pytest.org/en/latest/reference.html#pytest-approx) is a way to remove that assumption.

Reply:
For `1.25` that is not necessary since it is a number which can be exactly represented as a floating point number (instead of e.g. `1.2`).